parse.cpp

```diff
@@ -36,16 +38,26 @@ void factor ();
 void factor_tail ();
 void add_op ();
 void mul_op ();
+void r_op(); // TODO: Figure out what r stands for
+void condition();
+
+// TODO: Delete "predict program" print statements when no longer useful
 
 void program () {
     switch (input_token) {
         case t_id:
+            stmt_list();
+            break;
         case t_read:
+            stmt_list();
+            break;
         case t_write:
+            stmt_list();
+            break;
         case t_eof:
             printf ("predict program --> stmt_list eof\n");
             stmt_list ();
-            match (t_eof);
+            match (t_eof); // Do we only match when we consume a terminal in the line of the grammar we're in the method of?
```

Comment:
No this is just an example of a state's execution. 

We'll need to design on paper DFA / Pushdown DFA for CFG (He might have given us the full design not totally sure if we need to add to it) but, each case in this list represents a node in the DFA,  inside each case will consist of the logic that defines their possible transitions.